resources/assets/lib/play-detail-menu.coffee

```diff
@@ -0,0 +1,82 @@
+###
+#    Copyright 2015-2018 ppy Pty. Ltd.
+#
+#    This file is part of osu!web. osu!web is distributed with the hope of
+#    attracting more community contributions to the core ecosystem of osu!.
+#
+#    osu!web is free software: you can redistribute it and/or modify
+#    it under the terms of the Affero GNU General Public License version 3
+#    as published by the Free Software Foundation.
+#
+#    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+#    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#    See the GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+###
+
+import { createElement as el, createRef, PureComponent } from 'react'
+import { a, button, div, i } from 'react-dom-factories'
+
+export class PlayDetailMenu extends PureComponent
+  constructor: (props) ->
+    super props
+
+    @uuid = osu.uuid()
+    @menu = createRef()
+
+    @state =
+      active: false
+
+
+  componentDidMount: =>
+    $(document).on "click.#{@uuid}", @hide
+    $(document).on "keyup.#{@uuid}", @hide
+
+
+  componentWillUnmount: =>
+    $(document).off ".#{@uuid}", @hide
+
+
+  hide: (e) =>
```

Comment:
should add `@state.active` check otherwise all other instances will run `@props.onHide` when clicking inside a menu.

Reply:
previously every instance had its own listener without jquery, so this wasn't a problem. Then I decided to have the event handling multiplexed into a single listener through jquery, but seeing as it triggers every handler, I'm not sure it's such a good idea anymore :s

Reply:
NEVERMIND :best: